src/devices/Gpio/Drivers/Sunxi/SunxiDriver.cs

```diff
@@ -0,0 +1,476 @@
+                case PinMode.InputPullDown:
+                case PinMode.InputPullUp:
+                case PinMode.Output:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the mode of a pin.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
+        /// <returns>The mode of the pin.</returns>
+        protected override PinMode GetPinMode(int pinNumber)
+        {
+            if (!_pinModes.ContainsKey(pinNumber))
+            {
+                throw new InvalidOperationException("Can not get a pin mode of a pin that is not open.");
+            }
+
+            return _pinModes[pinNumber].CurrentPinMode;
+        }
+
+        /// <inheritdoc/>
+        protected override void Dispose(bool disposing)
+        {
+            if (_gpioPointer0 != default)
+            {
+                Interop.munmap(_gpioPointer0, 0);
+                _gpioPointer0 = default;
+            }
+
+            if (_gpioPointer1 != default)
+            {
+                Interop.munmap(_gpioPointer1, 0);
+                _gpioPointer1 = default;
+            }
+
+            if (_interruptDriver != default)
+            {
+                _interruptDriver.Dispose();
+                _interruptDriver = default;
+            }
+        }
+
+        private void InitializeInterruptDriver()
+        {
+            try
+            {
+                _interruptDriver = new LibGpiodDriver(0);
+            }
+            catch (PlatformNotSupportedException)
+            {
+                _interruptDriver = new InterruptSysFsDriver(this);
+            }
+        }
+
+        private void Initialize()
+        {
+            if (_gpioPointer0 != default)
+            {
+                return;
+            }
+
+            lock (s_initializationLock)
+            {
+                if (_gpioPointer0 != default)
+                {
+                    return;
+                }
+
+                int fileDescriptor = Interop.open(GpioMemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
+                if (fileDescriptor == -1)
+                {
+                    throw new IOException($"Error {Marshal.GetLastWin32Error()} initializing the Gpio driver.");
+                }
+
+                IntPtr mapPointer0 = Interop.mmap(IntPtr.Zero, Environment.SystemPageSize - 1, (MemoryMappedProtections.PROT_READ | MemoryMappedProtections.PROT_WRITE), MemoryMappedFlags.MAP_SHARED, fileDescriptor, CpuxPortBaseAddess & ~_mapMask);
+                IntPtr mapPointer1 = Interop.mmap(IntPtr.Zero, Environment.SystemPageSize - 1, (MemoryMappedProtections.PROT_READ | MemoryMappedProtections.PROT_WRITE), MemoryMappedFlags.MAP_SHARED, fileDescriptor, CpusPortBaseAddess & ~_mapMask);
+                if (mapPointer0.ToInt64() == -1 || mapPointer1.ToInt64() == -1)
```

Comment:
should you unmap in here if one of them is -1 and the other one isn't?

Reply:
(might be a good idea to create SafeMappedMemoryHandle or something to take care of this automatically)